src/common/k8s-api/endpoints/horizontal-pod-autoscaler.api.ts

```diff
@@ -263,3 +209,107 @@ export class HorizontalPodAutoscalerApi extends KubeApi<HorizontalPodAutoscaler>
     });
   }
 }
+
+
+
+function getMetricName(metric: HorizontalPodAutoscalerMetricSpec | HorizontalPodAutoscalerMetricStatus): string | undefined {
+  switch (metric.type) {
+    case HpaMetricType.Resource:
+      return metric.resource.name;
+    case HpaMetricType.Pods:
+      return metric.pods.metricName;
+    case HpaMetricType.Object:
+      return metric.object.metricName;
+    case HpaMetricType.External:
+      return metric.external.metricName;
+    case HpaMetricType.ContainerResource:
+      return metric.containerResource.name;
+    default:
+      return undefined;
+  }
+}
+
+function getResourceMetricValue(currentMetric: ResourceMetricStatus | undefined, targetMetric: ResourceMetricSource): MetricCurrentTarget {
+  return {
+    current: (
+      currentMetric?.currentAverageUtilization
+        ? `${currentMetric.currentAverageUtilization}%`
+        : currentMetric?.currentAverageValue
+    ),
+    target: (
+      targetMetric?.targetAverageUtilization
+        ? `${targetMetric.targetAverageUtilization}%`
+        : targetMetric?.targetAverageValue
+    ),
+  };
+}
+
+function getPodsMetricValue(currentMetric: PodsMetricStatus | undefined, targetMetric: PodsMetricSource): MetricCurrentTarget {
+  return {
+    current: currentMetric?.currentAverageValue,
+    target: targetMetric?.targetAverageValue,
+  };
+}
+
+function getObjectMetricValue(currentMetric: ObjectMetricStatus | undefined, targetMetric: ObjectMetricSource): MetricCurrentTarget {
+  return {
+    current: (
+      currentMetric?.currentValue
+        ?? currentMetric?.averageValue
+    ),
+    target: (
+      targetMetric?.targetValue
+        ?? targetMetric?.averageValue
+    ),
+  };
+}
+
+function getExternalMetricValue(currentMetric: ExternalMetricStatus | undefined, targetMetric: ExternalMetricSource): MetricCurrentTarget {
+  return {
+    current: (
+      currentMetric?.currentValue
+        ?? currentMetric?.currentAverageValue
+    ),
+    target: (
+      targetMetric?.targetValue
+        ?? targetMetric?.targetAverageValue
+    ),
+  };
+}
+
+function getContainerResourceMetricValue(currentMetric: ContainerResourceMetricStatus | undefined, targetMetric: ContainerResourceMetricSource): MetricCurrentTarget {
+  return {
+    current: (
+      currentMetric?.currentAverageUtilization
```

Comment:
nit: if `currentMetric.currentAverageUtilization === 0` shouldn't it really return `0%`?

Reply:
Yes it should. Will fix in a follow up PR.